crates/spk-solve/src/error.rs

```diff
@@ -65,6 +66,10 @@ pub enum Error {
     SolverLogFileIOError(#[source] std::io::Error, PathBuf),
     #[error("Error: Flushing solver log file: {0}")]
     SolverLogFileFlushError(#[source] std::io::Error),
+    #[error(
+        "Cannot build package ({0}) from source during a solve when it has a dependency on itself"
+    )]
+    SolverBuildFromSourceDependencyLoopError(VersionIdent),
```

Comment:
This feels like it would be worth having a documented error code for since additional explanation and context and help might be useful
```suggestion
    #[diagnostic(
        code(spk::solve::recursive_build_dependency),
        help("this package requires an older version of itself to build but none exists")
    )
    SolverBuildFromSourceDependencyLoopError(VersionIdent),
```

Reply:
I've updated this.